verticapy/machine_learning/metrics/classification.py

```diff
@@ -1854,6 +1915,43 @@ def roc_auc_score(
             fun_sql_name="roc",
         )[1:]
         return _compute_area(true_positive, false_positive)
+    elif average == "micro":
+        _check_labels(y_true=y_true, labels=labels, input_relation=input_relation)
+        _, false_positive, true_positive = _compute_micro_multiclass_metric(
+            y_true, y_score, input_relation, labels, nbins, fun_sql_name="roc"
+        )
+        return _compute_area(true_positive, false_positive)
+    elif average == "macro":
+        _check_labels(y_true=y_true, labels=labels, input_relation=input_relation)
+        fpr = {}
+        tpr = {}
+        for i in range(len(y_score)):
+            fpr[i], tpr[i] = _compute_function_metrics(
+                y_true=y_true,
+                y_score=y_score[i],
+                input_relation=input_relation,
+                pos_label=labels[i],
+                nbins=nbins,
+                fun_sql_name="roc",
+            )[1:]
+
+        fpr_grid = np.linspace(0.0, 1.0, nbins)
+        mean_tpr = np.zeros_like(fpr_grid)
+
+        for i in range(len(y_score)):
+            sorted_pairs = sorted(zip(fpr[i], tpr[i]))
+            fpr_sorted, tpr_sorted = zip(*sorted_pairs)
+            mean_tpr += np.interp(
+                fpr_grid, fpr_sorted, tpr_sorted
+            )  # linear interpolation
+
+        # Average it and compute AUC
+        mean_tpr /= len(y_score)
+
+        fpr_macro = fpr_grid
```

Comment:
why this variable creation?

Reply:
only for understanding. that the now i am calculating the "macro" using FPR and TPR.